### PR TITLE
task_wdt: Fix build warning on 64-bit platforms

### PR DESCRIPTION
**Pull request by pdgendt**
When building for 64-bit platform (tested on native_posix_64), there was a cast warning for different sizes.
A void * is used to store a simple ID, fixed by using uintptr_t instead.

### CODE REVIEW

**Review by martinjaeger — CHANGES_REQUESTED** [review]
`uintptr_t` doesn't look right for this purpose, as the channel ID is not a pointer. Why not use `size_t` everywhere?

**Comment by pdgendt** [comment]
> `uintptr_t` doesn't look right for this purpose, as the channel ID is not a pointer. Why not use size_t everywhere?

Not sure I agree, the documentation states:

>Integer type capable of holding a value converted from a void pointer and then be converted back to that type with a value that compares equal to the original pointer.

Which is what is done here. I think this is a portable way without the need to use `sizeof` calls.

**Review by martinjaeger — DISMISSED** [review]
Ok, with that definition it makes sense. Somehow found the statement `uintptr_t next_channel_id` confusing because it's not a pointer.

There is also an `intptr_t` which would allow the negative value as used before, but `UINTPTR_MAX` is also fine.

File: subsys/task_wdt/task_wdt.c [hunk 1]
```diff
@@ -65,7 +65,7 @@ static void schedule_next_timeout(int64_t current_ticks)
 #endif
 
 	/* find minimum timeout of all channels */
-	for (int id = 0; id < ARRAY_SIZE(channels); id++) {
+	for (size_t id = 0; id < ARRAY_SIZE(channels); id++) {
```
Comment (anangl) [inline subsys/task_wdt/task_wdt.c:68]: Is this change actually needed (it does not seem to be related to some build warning)? If so, there is another `for` loop below that also uses an `int` iterator, so it would be good to keep both consistent.

**Review by anangl — DISMISSED** [review]

**Review by martinjaeger — APPROVED** [review]

**Review by anangl — APPROVED** [review]
